apps/ledger-live-mobile/src/components/SetEnvsFromSettings.ts

```diff
@@ -1,45 +1,25 @@
-import { PureComponent } from "react";
-import { connect } from "react-redux";
-import { createStructuredSelector } from "reselect";
 import { setEnvUnsafe } from "@ledgerhq/live-env";
+import { useEffect } from "react";
+import { useSelector } from "react-redux";
 import {
-  hideEmptyTokenAccountsEnabledSelector,
   filterTokenOperationsZeroAmountEnabledSelector,
+  hideEmptyTokenAccountsEnabledSelector,
 } from "../reducers/settings";
-import { State } from "../reducers/types";
 
-const mapStateToProps = createStructuredSelector<
-  State,
-  {
-    hideEmptyTokenAccountsEnabled: boolean;
-    filterTokenOperationsZeroAmountEnabled: boolean;
-  }
->({
-  hideEmptyTokenAccountsEnabled: hideEmptyTokenAccountsEnabledSelector,
-  filterTokenOperationsZeroAmountEnabled: filterTokenOperationsZeroAmountEnabledSelector,
-});
+const SetEnvsFromSettings = () => {
+  const hideEmptyTokenAccountsEnabled = useSelector(hideEmptyTokenAccountsEnabledSelector);
+  const filterTokenOperationsZeroAmountEnabled = useSelector(
+    filterTokenOperationsZeroAmountEnabledSelector,
+  );
 
-class SetEnvsFromSettings extends PureComponent<{
-  hideEmptyTokenAccountsEnabled: boolean;
-  filterTokenOperationsZeroAmountEnabled: boolean;
-}> {
-  apply() {
-    const { hideEmptyTokenAccountsEnabled, filterTokenOperationsZeroAmountEnabled } = this.props;
+  const apply = () => {
     setEnvUnsafe("HIDE_EMPTY_TOKEN_ACCOUNTS", hideEmptyTokenAccountsEnabled);
     setEnvUnsafe("FILTER_ZERO_AMOUNT_ERC20_EVENTS", filterTokenOperationsZeroAmountEnabled);
-  }
-
-  componentDidMount() {
-    this.apply();
-  }
+  };
 
-  componentDidUpdate() {
-    this.apply();
-  }
+  useEffect(() => apply());
```

Comment:
Don't we want to setEnvUnsafe when `hideEmptyTokenAccountsEnabled` or `filterTokenOperationsZeroAmountEnabled` changed ?
I think that we could have one useEffect for both with each value in the dependency array, wdyt?

Reply:
Yeah that'd make more sense !